File: Data Structures & Algorithms/valid-sudoku/submission-2.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        unordered_set<char> rows[9];
        unordered_set<char> cols[9];
        unordered_set<char> boxes[9];

        for (int row = 0; row < board.size(); ++row) {
            for (int col = 0; col < board.size(); ++col) {
                if (board[row][col] == '.') continue;
                char ch = board[row][col];
                int box = row / 3 * 3 + col / 3;
                if (rows[row].count(ch) || cols[col].count(ch) || boxes[box].count(ch)) {
                    return false;
                }
                rows[row].insert(ch);
                cols[col].insert(ch);
                boxes[box].insert(ch);
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/max-area-of-island/submission-2.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        directions = [[1, 0], [-1, 0], [0, 1], [0, -1]]
        maxArea = 0
        rows, cols = len(grid), len(grid[0])

        def deleteIsland(r, c):
            grid[r][c] = 0
            area = 1
            for direction in directions:
                row = r + direction[0]
                col = c + direction[1]
                if row < 0 or len(grid) <= row or col < 0 or len(grid[0]) <= col:
                    continue
                elif grid[row][col] == 1:
                    area += deleteIsland(row, col)
            return area

        for r in range(rows):
            for c in range(cols):
                if grid[r][c] == 1:
                    maxArea = max(maxArea, deleteIsland(r, c))

        return maxArea